### PR TITLE
2PC Fermi catalog

### PR DESCRIPTION
**Pull request by MRegeard**
Awaiting merge of #5056. 

Split of PR https://github.com/gammapy/gammapy/pull/4699. Here is the 2PC part.

### CODE REVIEW

**Comment by registerrier** [comment]
Could you rebase @MRegeard ?

**Comment by registerrier** [comment]
Can you rebase and update now that 3PC has been merged @MRegeard ?

**Comment by MRegeard** [comment]
@registerrier, it's done. I have to check that everything is fine because the rebase was a mess !

File: gammapy/catalog/fermi.py [hunk 1]
```diff
@@ -1686,6 +1884,56 @@ def __init__(self, filename="$GAMMAPY_DATA/catalogs/fermi/gll_psch_v13.fit.gz"):
...
 
+class SourceCatalog2PC(SourceCatalog):
+    """Fermi-LAT 2nd pulsar catalog.
+
+    - https://ui.adsabs.harvard.edu/abs/2013ApJS..208...17A
+    - https://fermi.gsfc.nasa.gov/ssc/data/access/lat/2nd_PSR_catalog/
+
+    One source is represented by `~gammapy.catalog.SourceCatalogObject2PC`.
+
+    TODO  : Fix the UnitsWarning here ...
```
Comment (QRemy) [inline gammapy/catalog/fermi.py:1891]: could you make the TODO more explicit and move it within the code as comment, otherwise it will show in the doc.

**Review by QRemy — COMMENTED** [review]
Thanks @MRegeard, I have just one minor comment :

File: gammapy/catalog/tests/test_fermi.py [hunk 2]
```diff
@@ -665,6 +785,8 @@ def test_name(self):
         assert self.source.name == self.source_name
 
     def test_row_index(self):
+        assert self.source.row_index == 26
+
```
Comment (QRemy) [inline gammapy/catalog/tests/test_fermi.py:789]:
```suggestion
```

File: gammapy/catalog/fermi.py [hunk 3]
```diff
@@ -1326,6 +1328,200 @@ def spatial_model(self):
         return model
 
 
+class SourceCatalogObject2PC(SourceCatalogObjectFermiPCBase):
```
Comment (QRemy) [inline gammapy/catalog/fermi.py:1331]:
```suggestion
class SourceCatalogObject2PC(SourceCatalogObjectFermiPCBase):
"""One source from the 2PC catalog.
```

File: gammapy/catalog/fermi.py (hunk 1 above)
Comment (MRegeard) [reply gammapy/catalog/fermi.py]: It is fixed, I removed it !

**Review by QRemy — APPROVED** [review]
Thanks @MRegeard, no further comment.

**Review by bkhelifi — APPROVED** [review]
Thanks a lot @MRegeard

**Pull request merged**